### DISCUSSION TITLE
Better integration with Mesh

### DISCUSSION THREAD

**Discussion by jhaley-ut** [graphql-hive/console#237] [Feature request]
Hey folks,

I'd love to see more integration with the features in GraphQL Mesh. We're looking at Mesh as an option for our gateway now because we have a variety of independent services that will be contributing to the unified graph - some ReST, some GraphQL. The ability for those individual service teams to register their contributing API specs (GraphQL or OAS), see the generated schema, apply transforms and see how the resulting unified schema would look would be amazing! Including that with your other validation and monitoring capabilities, along with the ability for a Mesh gateway to then pull the unified schema would make for a very compelling solution.

The other feature we'd love to see is the ability to manage the allowlisting of persisted queries in Hive in order to better secure our API and have that also be integrated with the Mesh Gateway for enforcement.

Thanks and keep up the great work!
Jason

**Comment by n1ru4l** [comment]
@jhaley-ut The gateway part of GraphQL Mesh is now Hive Gateway with great Hive schema registry integration by default, you can learn more here: https://the-guild.dev/graphql/hive/product-updates/2024-09-09-hive-gateway

Using the Hive schema registry for managing persisted documents is a feature we recently introduced through app deployments: https://the-guild.dev/graphql/hive/product-updates/2024-07-30-persisted-documents-app-deployments-preview
